pykickstart/commands/timesource.py

```diff
@@ -0,0 +1,146 @@
+        self.nts = kwargs.get("nts", False)
+
+    def __eq__(self, y):
+        if not y:
+            return False
+
+        if self.ntp_server:
+            return self.ntp_server == y.ntp_server and self.nts == y.nts
+        elif self.ntp_pool:
+            return self.ntp_pool == y.ntp_pool and self.nts == y.nts
+
+    def __ne__(self, y):
+        return not self == y
+
+    def __str__(self):
+        retval = BaseData.__str__(self)
+
+        args = self._getArgsAsStr()
+        if args:
+            retval += "timesource%s\n" % args
+
+        return retval
+
+    def _getArgsAsStr(self):
+        retval = ""
+
+        if self.ntp_server:
+            retval += " --ntp-server=%s" % self.ntp_server
+        if self.ntp_pool:
+            retval += " --ntp-pool=%s" % self.ntp_pool
+        if self.nts:
+            retval += " --nts"
+        if self.ntp_disable:
+            retval += " --ntp-disable"
+
+        return retval
+
+
+class F33_Timesource(KickstartCommand):
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+
+    def __init__(self, writePriority=0, *args, **kwargs):
+        KickstartCommand.__init__(self, writePriority, *args, **kwargs)
+        self.op = self._getParser()
+        self.timesource_list = kwargs.get("timesource_list", [])
+        self.exclusive_required_options = [("ntp_server", "--ntp_server"),
+                                           ("ntp_pool", "--ntp_pool"),
+                                           ("ntp_disable", "--ntp_disable")]
+
+    def __str__(self):
+        retval = ""
+        for timesource_command in self.timesource_list:
+            retval += timesource_command.__str__()
+
+        return retval
+
+    def _getParser(self):
+        op = KSOptionParser(prog="timesource", description="""
+                            Configures a timesource.""",
+                            version=F33)
+        op.add_argument("--ntp-server", version=F33, help="""
+                        A single NTP server.
+
+                        ``--ntp-server`` and ``--ntp-pool`` are mutually exclusive.
+                        """,
+                        metavar="")
+        op.add_argument("--ntp-pool", version=F33, help="""
+                        A single NTP pool.
+
+                        ``--ntp-pool`` and ``--ntp-server`` are mutually exclusive.
+                        """)
+        op.add_argument("--ntp-disable", action="store_true",
+                        default=False, version=F33, help="""
+                        If specified, disable any NTP based time sync,
+                        both on target system as well as in installation
+                        environment.
+                        """)
+        op.add_argument("--nts", dest="nts", action="store_true",
+                        default=False, version=F33, help="""
+                        If specified, consider the server/pool to be
```

Comment:
I think using `source` instead of the three `server/pool` makes it easier to read this.

Reply:
Maybe even "host"?

Reply:
I think "host" sounds good for the first one, "time source" for the second and third.

Fixed up accordingly, let me know what you think. :)